### ISSUE TITLE
[FTE2] " As a user who’d like to use the lighthouse app, following the  quick overview, it must seek my permission to access the camera "

### ISSUE THREAD

**Issue by Yoric** [project-lighthouse/camera-app.ios#19]

**Comment by sfoster** [comment]
I don't think this is related to the FTE overview. We just want to ask permission before camera first use. If the tour is skipped, or however the user gets there, if we don't already have permission to use the camera we should ask for it. We would want the same interaction if the user had previously used the app and since removed the camera permission

**Comment by sfoster** [comment]
Is there any scenario in which the app is functional without camera permission?

**Comment by azasypkin** [comment]
The only case where we don't need camera permission I can think of is `db management`, but I'm not sure if it sounds like a real one:

* Iterate through known items and replay voice labels;
* Re-record new voice labels (needs mic permission);
* Delete items from the DB (purge entire DB to save space);

Whether FTE asks for camera permission or not (i'd just warn user during FTE that he will need to give permission for mic and camera), it should not be the only place, since later on user can somehow revoke that permission and we should ask for it once again once Lighthouse needs camera.

**Comment by punamdahiya** [comment]
My understanding as per IA https://drive.google.com/drive/folders/0ByeF_fkExK_BandfNkUtU2NJVkU, since our main screen after launch screen (even if we skip FTE) is view finder we need to check and ask for camera permission there

**Comment by hkoka** [comment]
Good discussion. Camera permissions should be checked before displaying the viewfinder. User could have landed on the "Camera View" screen from FTE overview or directly from launch screen or from another entrypoint in the image capture/recognition flows. I think it will help if we start grooming these stories and writing down the criteria considering specific cases so its clear to implement and test.

**Comment by punamdahiya** [comment]
WIP https://github.com/project-lighthouse/camera-app.ios/pull/45

**Comment by punamdahiya** [comment]
![img_1696](https://cloud.githubusercontent.com/assets/597734/22343306/7e5d9aca-e3ac-11e6-8858-e0b68b965fd6.PNG)

**Comment by punamdahiya** [comment]
![img_1695](https://cloud.githubusercontent.com/assets/597734/22343320/8d8d2588-e3ac-11e6-8f43-010351a41e01.PNG)
